kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -386,11 +383,25 @@ private long doCommit(
     resolvedDomainMetadatas.forEach(
         dm -> metadataActions.add(createDomainMetadataSingleAction(dm.toRow())));
 
-    try (CloseableIterator<Row> stageDataIter = dataActions.iterator()) {
+    try (CloseableIterator<Row> userStageDataIter = dataActions.iterator()) {
+
+      CloseableIterator<Row> completeFileActionIter;
+      if (isReplaceTable()) {
+        // For now, block CTAS with replace; this will be unblocked in a follow-up PR
```

Comment:
What does `block CTAS with replace`? the "with" confuses me -- do you mean "block CTAS and replace"? Do you mean "block replace table as select? RTAS??